semantic/poly_types.go

```diff
@@ -822,6 +837,40 @@ func (k ObjectKind) resolvePolyType(kinds map[Tvar]Kind) (PolyType, error) {
 	return NewObjectPolyType(properties, k.lower, k.upper), nil
 }
 
+// NullableKind indicates that it is possible for this
+// variable to be the null type if no other type is
+// more appropriate.
+type NullableKind struct {
+	T PolyType
+}
+
+func (NullableKind) MonoType() (Type, bool)        { return Nil, true }
+func (NullableKind) freeVars(*Constraints) TvarSet { return nil }
+func (n NullableKind) resolveType(kinds map[Tvar]Kind) (Type, error) {
+	return Nil, nil
```

Comment:
Is it possible for resolveType to be called on a NullableKind whose T value points to an integer or some concrete type?

Reply:
I don't think it is but I spent a bunch of time trying to figure that out. I believe that if this resolves to a concrete type, then you get that type directly instead of the `Tvar` and the only time kinds are consulted is when you have a `Tvar`.

Reply:
This is again completely correct. Resolving this conversation.